### PR TITLE
Advanced Building Blocks - Enumerables

### PR DESCRIPTION
**Pull request by verissimohenry**
In this project, I will implement my own version of Ruby’s enumerable methods.

### CODE REVIEW

**Review by MkrtichSargsyan — CHANGES_REQUESTED** [review]
## _Status:_ Invalid submission warning

## Hello Verissimo Henry


This Code Review Request will be marked as invalid in your Dashboard because your enumerable functions are not working, you have to put all of them inside ```module Enumerable```.

Please make the necessary changes and submit another code review request once you are ready! :slightly_smiling_face:.


> ps: . If you have any questions about my review or need help, you can message me on slack ( @mkrtichsargsyan )

File: README.md [hunk 1]
```diff
@@ -29,26 +33,28 @@ Clone repository from github to your local drive by following these steps:
...
 
 ### Install
+
 - Install Atom or Microsoft Visual Studio text editor to view the code on your local machine.
 - Install Git.
 
 ### Deployment
+
 - Open live-server from your text-editor to see the website.
 - Use github pages to view from the Live Demo link above.
```
Comment (Berabjesus) [inline README.md]: - [ ] There is no deployment here so remove this portion from your readme file.

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,112 @@
+module Enumerable
+  def my_each
+    i = 0
+    return enum unless block_given?
```
Comment (Berabjesus) [inline enumerable.rb]: - [ ] `my_each` returns an Enumerator if no block is given. Currently, its throwing the error below.
 image

File: enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    while to_a.length > i
+      yield(to_a[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
```
Comment (Berabjesus) [inline enumerable.rb:13]: - [ ] `my_each_with_index` returns an Enumerator if no block is given. Currently, its throwing the error below.
image

File: enumerable.rb [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    self
+  end
+
+  def my_select
+    new_array = []
+    return enum unless block_given?
+
+    to_a.my_each { |item| new_array.push(item) if yield(item) }
+    new_array
+  end
```
Comment (Berabjesus) [inline enumerable.rb]: - [ ] `my_select` returns an Enumerator if no block is given. Currently, its throwing the error below.
image

**Review by Berabjesus — CHANGES_REQUESTED** [review]
## Status:  CHANGES REQUIRED :warning: :construction: 
Hi @verissimohenry a very awesome start :+1: Good job with the changes you made after the last invalid review. 👏 

### However, you do have a few changes to carry out before we proceed so please take a look at the things you have to work on and their respective review comments below.

- [x] should have working `my_each`  identical and returns the same thing as ruby's `each`. ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))
- [x] should have working `my_each_with_index` identical and returns the same thing as ruby's `each_with_index`. ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))
- [x] should have working `my_select` identical and returns the same thing as ruby's `select`. ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))
- [x] Remove unnecessary depolyment section from readme

**Please have a look at the specific comments below**

### **Please, do not open a new Pull Request for subsequent reviews. You should use the same Pull Request submitted for the first review unless it is requested otherwise.**

#### **If you have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help** 
**HAPPY CODING :computer:[Bereket Beshane](https://bereketababu.me)** | [LinkedIn](https://linkedin.com/in/bereketbeshane) | [GitHub](https://github.com/Berabjesus)

**Review by jubaan — APPROVED** [review]
[![Microverse][microverse-shield]][microverse-url]
<div>
<h1 align="center"> 
  🎉🥳 🎊 🎉 🥳 🥳 🎊 🎉 🥳🥳 🎊 🎉 🥳 🎉 <br />
  🎉🥳 🎊 🎉 🥳 🥳 🎊 🎉 🥳🥳 🎊 🎉 🥳 🎉 <br />
  🥳 🎊 🎉 🥳  APPROVED 🥳 🎊 🎉 🥳 <br />
  🎉🥳 🎊 🎉 🥳 🥳 🎊 🎉 🥳🥳 🎊 🎉 🥳 🎉 <br />
  🎉🥳 🎊 🎉 🥳 🥳 🎊 🎉 🥳🥳 🎊 🎉 🥳 🎉
</h1>
<h2 align="center">
  What’s done is done. What’s gone is gone. <br />
  One of life’s lessons is always moving on. <br />
  It’s okay to look back to see how far you’ve come but <br /> 
  KEEP MOVING FORWARD.
</h2>

<details open="open">
  <summary>Review Summary</summary>
  <ol>
    <li><a href="#optional"> 💡Nice to have and other comments ✍🏼 ✍🏼 </a></li>
    <li><a href="#contact"> 🧔🏼 About your Reviewer </a> (you don't know it... but we are just automated bots 🤖) </li>
    <li><a href="#help"> 🆘 <b> 😭 Feeling stuck? or Need help</b>? 🙏🏼 🙏🏼 </a></li>
  </ol>
</details>
</div>


<h3 id="optional"> 1. 💡 Nice to have and other comments </h3>
<details close="close">
<summary> [click to open] If you have something here, even when approved, it's just a little push to improve your project.</h3></summary>


- [ ] Nothing more to add.


</details>

<h3 id="contact"> 2. 🧔🏼 About your Reviewer </h3>
<details close="close">
<summary> [click to open] Say "hello world!" on Slack: @jubaan</summary>
<br />
<div align="center">
  <a src="https://www.github.com/jubaan">
    Julio Añoveros
  </a>
  <p>
    Julio Añoveros | @jubaan</br></br>
    <b>Full-stack Web Developer</b> and</br>
    🤖 Code Reviewer at <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a>:rocket:
  </p>
</div>
</details>

<h3 id="help"> 3. 🆘 <b> 😭 Feeling stuck? or Need help</b>? 🙏🏼 🙏🏼</h3>
<p> How does it feel? Is nice to advance in the program, right? It keeps you motivated even with all the struggles and takes you one step closer to the goal 🎯!. I wan't you to keep advancing so if you feel stuck and/or need help, remember there is always someone you can reach to keep moving forward, doesn't mater how "dum" you think the questions is.</p>
<details close="close">
  <summary> [click to open] <b>Feel stuck?</b> </summary>
  <br>
  <p> First breath... just do it. Take 15-20 min away from the computer, stretch, meditate, do some exercise, drink water, or do whatever is necessary to distract your mind from the obstacle.</p>
  <p><b>Sometimes you just need to get away to see things with a new perspective.</b></p>
</details>

<details close="close">
  <summary> [click to open] <b>Need help?</b> </summary>
  <br>
  <p> Send me a DM in <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> Students Slack, you'll find me as @jubaan. If I'm not available there are other Code Reviewers you can contact. They'll help as well.</p>
  <p>Remember that <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> has other channels where you can get an answer to your questions:</p>
  <ol>
    <li>The <b>students channel 🧑‍🎓 </b> in slack,</li>
    <li>your <b>teammates 👥 </b>,</li>
    <li>your <b>mentor 👨🏼 👩🏼 </b>, and</li>
    <li><b>Student Success 🏫 </b>.</li>
  </ol>
  <p><b> DON'T hesitate to ask! </b></p>
</details>
<p>If all of the above doesn't help... get a <a href="https://rubberduckdebugging.com/">rubber duck</a> ... it would be great to own one 😄. </p>

<h1 align="center"> See you a the top 🚀!! </h1>

[microverse-shield]: https://img.shields.io/badge/-microverse-%236F23FF?style=for-the-badge
[microverse-url]: https://www.microverse.org/?grsf=kfm2ej
[bootstrap-shield]: https://img.shields.io/badge/Bootstrap-white?style=for-the-badge&logo=bootstrap
[bootstrap-url]: https://www.microverse.org/?grsf=kfm2ej

**Pull request merged**